### PR TITLE
SDK-892 Change DOB from string to date

### PR DESCRIPTION
**Pull request by liamd-yoti**
`getDateOfBirth` now returns `Date` instead of string.

### CODE REVIEW

**Comment by abdalmajeed-yoti** [comment]
How about converting it to a `Date` object after it's converted in the `AttributeConverter` instead of converting it every time `getDateOfBirth` gets called? 
I think you can do that in `/yoti-node-sdk/src/yoti_common/converters/attribute.converter.js` line `51`.

**Comment by liamd-yoti** [comment]
> How about converting it to a `Date` object after it's converted in the `AttributeConverter` instead of converting it every time `getDateOfBirth` gets called? I think you can do that in `/yoti-node-sdk/src/yoti_common/converters/attribute.converter.js` line `51`.

Okay think I was confused about this. It would indeed work. However when I was running the unit-tests for `profile.spec.js`, they are actually not a source of truth for the data types as we construct just the `Profile` class which does not perform any sort of conversion itself. The conversion is only done when we decrypt some profile. So yeah just updated the tests for the converter so that we can confirm that the data type is indeed changed when we receive some date field.

**Review by abdalmajeed-yoti — APPROVED** [review]
Looks good to me. 👍

File: tests/yoti_common/converters/attribute.converter.spec.js [hunk 1]
```diff
@@ -239,6 +239,10 @@ describe('AttributeConverter', () => {
         expect(errMessage).toBe('Warning: value is NULL', `Content Type: ${contentType}`);
       });
     });
+    it('should return Date values', () => {
+      const converted = AttributeConverter.convertValueBasedOnContentType('1979-01-01', CONTENT_TYPE_DATE);
+      expect(converted).toBeInstanceOf(Date);
```
Comment (laurent-yoti) [inline tests/yoti_common/converters/attribute.converter.spec.js:244]: Probably good to check the date is matching ;-)
Please add a few lines to check the yeah, month...

**Review by abdalmajeed-yoti — APPROVED** [review]

**Review by laurent-yoti — APPROVED** [review]
Great, please merge.

**Pull request merged**